### PR TITLE
TST: add pattern for nearest

### LINKED ISSUE #33895: BUG: resample().nearest() throws an exception if tz is provided for the DatetimeIndex
Here a simple create of a pandas DataFrame:
```python
import pandas as pd
df = pd.DataFrame(range(5), index=pd.date_range('1/1/2000', periods=5, freq='H', tz='UTC'))
print(df) gives:
2000-01-01 00:00:00+00:00  0
2000-01-01 01:00:00+00:00  1
2000-01-01 02:00:00+00:00  2
2000-01-01 03:00:00+00:00  3
2000-01-01 04:00:00+00:00  4
```
When trying to resample like this
```python
df.resample('15Min').nearest()
```
this throws an exception:
```python
File "C:\Users\user\anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-41-9b9242799057>", line 1, in <module>
    df.resample('15Min').nearest()
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\resample.py", line 517, in nearest
    return self._upsample("nearest", limit=limit)
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\resample.py", line 1095, in _upsample
    res_index, method=method, limit=limit, fill_value=fill_value
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\util\_decorators.py", line 227, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\frame.py", line 3856, in reindex
    return self._ensure_type(super().reindex(**kwargs))
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\generic.py", line 4544, in reindex
    axes, level, limit, tolerance, method, fill_value, copy
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\frame.py", line 3744, in _reindex_axes
    index, method, copy, level, fill_value, limit, tolerance
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\frame.py", line 3760, in _reindex_index
    new_index, method=method, level=level, limit=limit, tolerance=tolerance
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\indexes\base.py", line 3145, in reindex
    target, method=method, limit=limit, tolerance=tolerance
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\indexes\base.py", line 2740, in get_indexer
    indexer = self._get_nearest_indexer(target, limit, tolerance)
  File "C:\Users\user\anaconda3\lib\site-packages\pandas\core\indexes\base.py", line 2821, in _get_nearest_indexer
    left_distances = abs(self.values[left_indexer] - target)
numpy.core._exceptions.UFuncTypeError: ufunc 'subtract' cannot use operands with types dtype('<M8[ns]') and dtype('O')
```

However, if we remove the **tz** filed it works:
```python
import pandas as pd
df = pd.DataFrame(range(5), index=pd.date_range('1/1/2000', periods=5, freq='H'))
df.resample('15Min').nearest()

OutPut: 
                     0
2000-01-01 00:00:00  0
2000-01-01 00:15:00  0
2000-01-01 00:30:00  1
2000-01-01 00:45:00  1
2000-01-01 01:00:00  1
2000-01-01 01:15:00  1
2000-01-01 01:30:00  2
2000-01-01 01:45:00  2
2000-01-01 02:00:00  2
2000-01-01 02:15:00  2
2000-01-01 02:30:00  3
2000-01-01 02:45:00  3
2000-01-01 03:00:00  3
2000-01-01 03:15:00  3
2000-01-01 03:30:00  4
2000-01-01 03:45:00  4
2000-01-01 04:00:00  4
```

The used pandas version is:
```
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 12, GenuineIntel

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
```

</details>

### PR DESCRIPTION
**Pull request by kotamatsuoka**
- [ ] closes #33895
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/resample/test_datetime_index.py [hunk 1]
```diff
@@ -535,11 +535,13 @@ def test_upsample_with_limit():
     tm.assert_series_equal(result, expected)
 
 
-def test_nearest_upsample_with_limit():
-    rng = date_range("1/1/2000", periods=3, freq="5t")
+@pytest.mark.parametrize("freq", ["Y", "10M", "5D", "10H", "5Min", "10S"])
```
Comment (jreback) [inline pandas/tests/resample/test_datetime_index.py:538]: you need to add the tz_aware_fixture here as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_datetime_index.py (hunk 1 above)
Comment (kotamatsuoka) [reply pandas/tests/resample/test_datetime_index.py:538]: Thanks @jreback. I've done.

File: pandas/tests/resample/test_datetime_index.py [hunk 2]
```diff
@@ -535,11 +535,13 @@ def test_upsample_with_limit():
...
+@pytest.mark.parametrize("rule", ["Y", "3M", "15D", "30H", "15Min", "30S"])
+def test_nearest_upsample_with_limit(tz_aware_fixture, freq, rule):
+    rng = date_range("1/1/2000", periods=3, freq=freq, tz=tz_aware_fixture)
```
Comment (mroeschke) [inline pandas/tests/resample/test_datetime_index.py:541]: Could you put the Github issue number here `# GH ...`, otherwise LGTM

Comment (kotamatsuoka) [reply pandas/tests/resample/test_datetime_index.py:542]: OK, I added!

**Review by mroeschke — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @kotamatsuoka

**Pull request merged**